src/components/paymentPage.js

```diff
@@ -1,61 +1,81 @@
-import { clearCart } from "../store";
 
 function PaymentPage() {
   const { state } = useLocation();
   const navigate = useNavigate();
   const dispatch = useDispatch();
   const itemList = useSelector((state) => state.cart.list);
   const userId = useSelector((state) => state.profile.id);
-  const isTotalPriceValid = state?.total > 0;
-  const isItemListValid = state?.itemList?.length > 0;
-  const [paymentInformation, setPaymentInformation] = useState({});
-  const [selectedPaymentInformation, setSelectedPaymentInformation] =
-    useState("");
-
-  const checkedHandler = (event) => {
-    if (event.target.checked) {
-      setSelectedPaymentInformation({ type: event.target.value });
-    }
-  };
+  const [selectedPaymentInformation, setSelectedPaymentInformation] = useState("");
+  const [bankFormValid, setBankFormValid] = useState(false);
+  const [addressFormValid, setAddressFormValid] = useState(false);
 
   useEffect(() => {
-    if (!isTotalPriceValid || !isItemListValid) {
+    if (!(state?.total > 0 && state?.itemList?.length > 0)) {
       navigate("/");
-      return;
     }
-  }, [isTotalPriceValid, isItemListValid, navigate]);
+  }, [state, navigate]);
 
-  useEffect(() => {
-    if (!paymentInformation.type || itemList.length < 1) {
+  const handlePayment = async () => {
+    if (!selectedPaymentInformation || itemList.length < 1 || !bankFormValid || !addressFormValid) {
+      // Mensaje de error usando SweetAlert
+      if (!selectedPaymentInformation) {
+        await Swal.fire({
+          icon: 'error',
+          title: '¡Error!',
+          text: 'Por favor, selecciona un método de pago.',
+        });
+      }
+      if (itemList.length < 1) {
+        await Swal.fire({
+          icon: 'error',
+          title: '¡Error!',
+          text: 'Tu carrito está vacío. Agrega elementos antes de realizar el pago.',
+        });
+      }
+      if (!bankFormValid || !addressFormValid) {
+        await Swal.fire({
+          icon: 'error',
+          title: '¡Error!',
+          text: 'Por favor, completa todos los campos en los formularios.',
+        });
+      }
       return;
     }
 
-    completePaymentOrderAsync(userId, paymentInformation, itemList);
-
-    Swal.fire("Compra completada!").then(() => {
+    try {
+      await completePaymentOrderAsync(userId, selectedPaymentInformation, itemList);
+      Swal.fire("¡Compra completada!");
       dispatch(clearCart());
       navigate("/");
-    });
-  }, [userId, paymentInformation, itemList, dispatch, navigate]);
+    } catch (error) {
+      console.error("Error al completar la compra:", error);
+      Swal.fire("Hubo un error al completar la compra.");
+    }
+  };
+
+  const checkedHandler = (event) => {
+    if (event.target.checked) {
+      setSelectedPaymentInformation({ type: event.target.value });
+    }
+  };
 
   return (
     <>
-      <header className="p-3 header_container w-100">
-        <Header />
-      </header>
+      <Header />
```

Comment:
Could you check this change is not breaking the header when the user is on /pay page?

Reply:
> Could you check this change is not breaking the header when the user is on /pay page?

ready!